src/coreclr/vm/common.h

```diff
@@ -367,6 +367,7 @@ inline void ClrRestoreNonvolatileContext(PCONTEXT ContextRecord)
 #include "pefile.inl"
 #include "excep.h"
 #include "method.hpp"
+#include "field.h"
```

Comment:
Which part of the change needs to have the field.h include here?

Reply:
This is required on linux x86, because `IsArgumentInRegister` accesses FieldDesc. Otherwise:

```
  /home/z/Dev/runtime3/src/coreclr/binder/../vm/callingconvention.h:416:46: error: member access into incomplete type 'FieldDesc'
                      CorElementType type = pFD->GetFieldType();

```

Reply:
Can you please instead move the `#include "frames.h"` above the `#include "field.h"` and revert the change in `field.h` and `field.cpp`? I've verified it compiles fine on x86 Linux that way.

Reply:
I still get the original build error this way, because `"frames.h"` itself includes `"callingconvention.h"`, which requires `"field.h"` on x86. So, if `"frames.h"` is above `"field.h"` there still is `error: member access into incomplete type 'FieldDesc'`

Reply:
Ah, I am sorry, I've found I've forgotten to add "cross" to my command line when trying to build for x86. Please keep it as you had it.